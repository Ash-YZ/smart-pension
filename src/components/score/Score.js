import React from 'react'
import PropTypes from 'prop-types'
import './styles.scss'

const Score = props => {
  return (
    <div className='score'>
      <h2>Score</h2>
      <div className='container'>
        {props.score.map((scr, idx) =>
          <div key={`score-${idx}`} className='row justify-content-center'>
            <>
              <div className='col-6 text-left'>
                <h2>Player {idx + 1}:</h2>
              </div>
              <div className='col-6 text-right'>
                <h2>{scr}</h2>
              </div>
            </>
          </div>
        )}
      </div>
    </div>
  )
}

Score.propTypes = {
  score: PropTypes.arrayOf(PropTypes.number)
}

export default Score
