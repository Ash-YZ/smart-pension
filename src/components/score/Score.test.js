import React from 'react';
import { shallow } from 'enzyme'
import Score from './Score'

const testProps = {
  score: [1, 2, 3]
}

describe('Score', () => {
  it('should render score correctly ', () => {
    const component = shallow(<Score {...testProps} />)
    expect(component).toMatchSnapshot()
  })
})