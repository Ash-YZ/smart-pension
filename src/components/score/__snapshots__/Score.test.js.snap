// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score should render score correctly  1`] = `
<div
  className="score"
>
  <h2>
    Score
  </h2>
  <div
    className="container"
  >
    <div
      className="row justify-content-center"
      key="score-0"
    >
      <div
        className="col-md-6 col-10 text-left"
      >
        <h2>
          Player 
          1
          :
        </h2>
      </div>
      <div
        className="col-md-5 col-2 text-right"
      >
        <h2>
          1
        </h2>
      </div>
    </div>
    <div
      className="row justify-content-center"
      key="score-1"
    >
      <div
        className="col-md-6 col-10 text-left"
      >
        <h2>
          Player 
          2
          :
        </h2>
      </div>
      <div
        className="col-md-5 col-2 text-right"
      >
        <h2>
          2
        </h2>
      </div>
    </div>
    <div
      className="row justify-content-center"
      key="score-2"
    >
      <div
        className="col-md-6 col-10 text-left"
      >
        <h2>
          Player 
          3
          :
        </h2>
      </div>
      <div
        className="col-md-5 col-2 text-right"
      >
        <h2>
          3
        </h2>
      </div>
    </div>
  </div>
</div>
`;
